components/shared/Select.tsx

```diff
@@ -0,0 +1,88 @@
+import { Fragment, useState, FC, SetStateAction } from 'react'
+import { Listbox, Transition } from '@headlessui/react'
+import { CheckIcon, ChevronUpDownIcon } from '@heroicons/react/20/solid'
+import { twMerge } from 'tailwind-merge'
+import classnames from 'classnames'
+
+interface SelectProps {
+  options: { id: number; name: string }[]
+  defaultSelected?: { id: number; name: string }
+  onSelect: (value: { id: number; name: string }) => void
+}
+
+export const tw: typeof classnames = (...params) =>
+  twMerge(classnames(...params))
+
+export const Select: FC<SelectProps> = ({
+  options,
+  defaultSelected,
+  onSelect,
+}) => {
+  const [selected, setSelected] = useState(defaultSelected || options[0])
+
+  const handleSelected = (value: { id: number; name: string }) => {
```

Comment:
quick tip: when working with typescript and react, it's always better to let typescript do the work for you. In this case, if you put this handler function inline (at line 29) you could've saved yourself from having to write the type for value since TS would've been able to imply it